### PR TITLE
[c10d] Add a timeout check interval variable for timeout dump

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117168
* __->__ #117093

The current timeout check frequency is relied on monitoring thread's timeout thread which can be too long (even if we set it to 2mins) so let's use a separate timeout variable which users can configure it. And we only only let default PG to check TCPStore so even more frequent check should be fine. (Our stress test is performed on every half second).

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -836,31 +838,30 @@ ProcessGroupNCCL::ProcessGroupNCCL(
...
-            << ", TORCH_NCCL_DESYNC_DEBUG: " << desyncDebug_
-            << ", TORCH_NCCL_ENABLE_TIMING: " << enableTiming_.load()
-            << ", TORCH_NCCL_BLOCKING_WAIT: " << blockingWait_
-            << ", TIMEOUT(ms): " << options_->timeout.count()
-            << ", USE_HIGH_PRIORITY_STREAM: "
-            << options_->is_high_priority_stream
-            << ", SPLIT_FROM: " << options_->split_from
-            << ", SPLIT_COLOR: " << options_->split_color
-            << ", TORCH_DISTRIBUTED_DEBUG: " << torch_distributed_debug
+  LOG(INFO)
```
Comment (fduwjj) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:841]: This is added by lintrunner automatically...

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp [hunk 2]
```diff
@@ -89,6 +89,11 @@ static std::vector<std::string> TORCH_NCCL_TRACE_BUFFER_SIZE = {
 static std::vector<std::string> TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC = {
     "TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC"};
 
+// Control the interval inside the watchdog thread to check the signal
+// from other timeout ranks to dump the debugging information.
+static std::vector<std::string> TORCH_NCCL_TIMEOUT_CHECK_MILSEC = {
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:94]: this name is a bit confusing (We have several other timeout variables/envs).

TORCH_NCCL_COORDINATED_DUMP_INTERVAL_MS
something like that?

Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:94]: Maybe `TORCH_NCCL_COORD_DUMP_CHECK_MS`?

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 3]
```diff
@@ -746,6 +746,8 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       getCvarInt(TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC, 60 * 10 /*10 Mins*/);
   waitTimeoutDumpInMilSec_ =
       getCvarInt(TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC, 2000);
+  coordDumpCheckIntervalMilSec_ =
+      getCvarInt(TORCH_NCCL_COORD_DUMP_CHECK_MS, 1000);
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:750]: nit - i guess we should align on MILSEC or MS consistently.  I prefer MS but i see we already have _SEC and _MILSEC so we should probably stick with that and then if we want to change them all in another PR we can

**Review by wconstab — APPROVED** [review]

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp (hunk 3 above)
Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:750]: Sure... I just don't want to make the name too long.. And MILSEC is less ambiguous.

**Review by kwen2501 — APPROVED** [review]
LGTM.

**Comment by fduwjj** [comment]
I already rebased to the latest. I think the mergeability test is indeed flasky.

**Pull request closed without merging**